prototype/p2p/testing_utils.go

```diff
@@ -0,0 +1,43 @@
+package p2p
+
+import (
+	"crypto/rand"
+	"sync/atomic"
+)
+
+type fnCallStub struct {
+	called *int32 // act as bool
+	timesc *int32 // act as int
+}
+
+func newFnCallStub() *fnCallStub {
+	return &fnCallStub{
+		called: new(int32),
+		timesc: new(int32),
+	}
+}
+
+func (f *fnCallStub) trackCall() {
+	atomic.AddInt32(f.called, 1)
+	atomic.AddInt32(f.timesc, 1)
+}
+
+func (f *fnCallStub) wasCalled() bool {
```

Comment:
I really like the idea of build custom tooling for our needs, but this seems to be something generic.

For example, if we use the `mockgen` library, we can literally tell it how many times it should be called and that way we are not rolling our own: 

```
	p2pNetworkMock.EXPECT().
		GetAddrBook().
		DoAndReturn(func() []*p2p_types.NetworkPeer {
			log.Println("[MOCK] Network GetNetwork", addrBook)
			return addrBook
		}).
		AnyTimes()
```		

Example: https://github.com/pokt-network/consensus-pre-prototype/blob/c54a9a062a7507639d30b6f920698c3d08605ffa/pkg/pocket/consensus_tests/utils_test.go#L32

We can discuss this in more detail

Reply:
I am interested in mockgen at this point. I've resorted to making my own "function call stub/spy" due to my ignorance of existing tools (_primarily_).